src/SockPerf.cpp

```diff
@@ -3176,20 +3168,27 @@ int bringup(const int *p_daemonize)
 
 #ifdef  USING_VMA_EXTRA_API
 	if ( !rc &&
-			(s_user_params.is_vmarxfiltercb || s_user_params.is_vmazcopyread || s_user_params.fd_handler_type == SOCKETXTREME)) {
+			(s_user_params.is_vmarxfiltercb ||
+					s_user_params.is_vmazcopyread ||
+					s_user_params.fd_handler_type == SOCKETXTREME)) {
 		// Get VMA extended API
 		g_vma_api = vma_get_api();
-		if (g_vma_api == NULL)
-			exit_with_err("VMA Extra API is required but not found", SOCKPERF_ERR_FATAL);
-		else
-			log_msg("VMA Extra API found - using VMA's receive zero copy and messages filter APIs");
+		if (g_vma_api == NULL) {
+			errno = EPERM;
+			exit_with_err("VMA Extra API is not available", SOCKPERF_ERR_FATAL);
+		} else {
+			log_msg("VMA Extra API is in use");
+		}
 
 		_vma_pkts_desc_size = sizeof(struct vma_packets_t) + sizeof(struct vma_packet_t) + sizeof(struct iovec) * 16;
 	}
 #else
 	if ( !rc &&
-	     		(s_user_params.is_vmarxfiltercb || s_user_params.is_vmazcopyread)) {
-		log_msg("This version is not compiled with VMA extra API");
+			(s_user_params.is_vmarxfiltercb ||
+					s_user_params.is_vmazcopyread ||
+					s_user_params.fd_handler_type == SOCKETXTREME)) {
+		errno = EPERM;
+		exit_with_err("Please compile with VMA Extra API to use these options", SOCKPERF_ERR_FATAL);
```

Comment:
It looks like we will quit while vmarxfiltercb is used but extra APi was not found,
Do we want to change the configure output ?
""If possible use VMA's receive path message filter callback API (See VMA's readme).""

https://github.com/Mellanox/sockperf/blob/sockperf_v2/src/SockPerf.cpp#L1439
same for vmazcopyread...

Reply:
Sorry, do not understand your concern.